tests/neptune/management/internal/test_utils.py

```diff
@@ -42,3 +47,51 @@ def test_normalize_project_name(self):
 
         with self.assertRaises(ConflictingWorkspaceName):
             normalize_project_name(name="jackie/sandbox", workspace="john")
+
+
+@pytest.mark.parametrize(
+    "test_input,expected",
+    [
+        (("01a", {"AAA", "BBB", "CCC"}), "01A"),
+        (("ddd", {"AAA", "BBB", "CCC"}), "DDD"),
+        (("aaaa", {"AAAA", "BBB", "CCC"}), "AAA"),
+        (("aa--aa", {"AAA", "BBB", "CCC"}), "AAAA"),
+        (("aaaa", {"AAA", "BBB", "CCC"}), "AAAA"),
+        (("aa---a", {"AAA", "BBB", "CCC"}), "AAA2"),
+        (("a-a---a--a", {"AAAA", "BBB", "CCC"}), "AAA"),
+        (("AAAA", {"AAA", "BBB", "CCC"}), "AAAA"),
+        (("aaaa", {"aaa", "BBB", "CCC"}), "AAAA"),
+        (("aaa3", {"AAA", "AAA2", "BBB", "CCC"}), "AAA3"),
+        (("aaaa", {"AAA", "AAA2", "BBB", "CCC"}), "AAAA"),
+        (("aaaa", {"AAA", "AAA2", "AAAA", "BBB", "CCC"}), "AAAA2"),
+        (("aaaa", {"AAA", "AAA2", "AAAA", "AAAA2", "BBB", "CCC"}), "AAA3"),
+        (("000-KEBAB", {"AAA", "BBB", "CCC"}), "000K"),
+        (("000-KEBAB", {"AAA", "BBB", "CCC", "000K"}), "000KE"),
+    ],
+)
+def test_project_key_simple_generation(test_input, expected):
+    result = ProjectKeyGenerator(test_input[0], test_input[1]).get_default_project_key()
+    assert expected == result
+
+
+@pytest.mark.parametrize(
+    "test_input,expected_prefix",
+    [
+        (("ccc", {"aaa", "bbb", "ccc", "ccc2", "ccc3"}), "CCC"),
+        (("aaaa", {"aaa", "aaa2", "aaa3", "aaaa", "aaaa2", "aaaa3", "bbb", "ccc"}), "AAAA"),
+        (("A----", {"AAA", "BBB", "CCC", "CCC2", "CCC3"}), "A"),
+        (("----", {"AAA", "BBB", "CCC", "CCC2", "CCC3"}), ""),
+        (("a-a", {"AAA", "BBB", "CCC", "CCC2", "CCC3"}), "AA"),
+        (("A-A", {"AAA", "BBB", "CCC", "CCC2", "CCC3"}), "AA"),
+        (("-0--1-----", {"AAA", "BBB", "CCC"}), "01"),
+        (("-000--1--", {"AAA", "BBB", "CCC"}), "0001"),
+        (("-000--111", {"AAA", "BBB", "CCC"}), "00011"),
+        (("-0--!@#$%^&*()_+=-[]{};':,./<>?|--1-", {"AAA", "BBB", "CCC"}), "01"),  # not the case now
```

Comment:
Key using only digits are invalid. So probably something like `01XYZ` (where XYZ are random) should be generated.

Reply:
Yes, it is done like this. The string on the right side is the prefix of the result.